### ISSUE TITLE
Support for AWS::CloudFormation::Interface

### ISSUE THREAD

**Issue by bjorg** [jakejscott/Humidifier#10]
Using `Humidifier v0.2.15`

It doesn't seem to be currently possible to generate a `Metadata`  stack entry. AFAIK, this is only used for the `AWS::CloudFormation::Interface` resource type which is handled in a really unusual manner.

```yaml
Metadata: 
  AWS::CloudFormation::Interface: 
    ParameterGroups: 
      - 
        Label: 
          default: "Network Configuration"
        Parameters: 
          - VPCID
          - SubnetId
          - SecurityGroupID
      - 
        Label: 
          default: "Amazon EC2 Configuration"
        Parameters: 
          - InstanceType
          - KeyName
    ParameterLabels: 
      VPCID: 
        default: "Which VPC should this be deployed to?"
```

See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudformation-interface.html

**Comment by bjorg** [comment]
For now, I'm working around it by deserializing the JSON template and adding the section manually. A tad hacky, but at least, I'm not blocked.

**Comment by jakejscott** [comment]
Thanks for reporting. At the moment there is an `AddMetadata` method for adding meta data to a resource which is a bummer cause that conflicts with adding template level metadata.

`AddMetadata` takes a resource name a dynamic metadata object (which is just json).

I could add another overload which would be `AddMetadata(dynamic metadata)` which would add metadata at the template (global) level.

**Comment by jakejscott** [comment]
Also the reason why I hadn't bothered to add support for this is because it is used only in the AWS console for making the UI a bit nicer to group parameters.

**Comment by jakejscott** [comment]
Fixed in 0.2.16 https://github.com/jakejscott/Humidifier/commit/be0c9c4a36367a045ef510f84820f7630033275d

**Comment by bjorg** [comment]
Thx for adding it!  Sorry for the long silence. I didn't get a notification from GitHub about the update.
